src/sentry/discover/endpoints/discover_key_transactions.py

```diff
@@ -2,29 +2,41 @@
 
 from django.db import transaction
 from rest_framework.response import Response
-from rest_framework.exceptions import ParseError
 
-from sentry import features
-from sentry.api.bases import OrganizationEventsV2EndpointBase
+from sentry.api.bases import KeyTransactionBase
 from sentry.api.bases.organization import OrganizationPermission
 from sentry.api.exceptions import ResourceDoesNotExist
 from sentry.discover.models import KeyTransaction
 from sentry.discover.endpoints.serializers import KeyTransactionSerializer
 from sentry.snuba.discover import query
 
 
-class KeyTransactionEndpoint(OrganizationEventsV2EndpointBase):
+class IsKeyTransactionEndpoint(KeyTransactionBase):
     permission_classes = (OrganizationPermission,)
 
-    def has_feature(self, request, organization):
-        return features.has("organizations:performance-view", organization, actor=request.user)
+    def get(self, request, organization):
+        """ Get the Key Transactions for a user """
+        if not self.has_feature(request, organization):
+            return self.response(status=404)
+
+        project = self.get_project(request, organization)
+
+        transaction = request.GET.get("transaction")
 
-    def get_project(self, request, organization):
-        projects = self.get_projects(request, organization)
+        try:
+            KeyTransaction.objects.get(
+                organization=organization,
+                owner=request.user,
+                project=project,
+                transaction=transaction,
+            )
+            return Response({"isKey": True}, status=200)
+        except KeyTransaction.DoesNotExist:
+            raise ResourceDoesNotExist
```

Comment:
@dashed dunno if this is more annoying for the frontend, can change this to `isKey: False` if its better for you.

Reply:
yea that might be better.